docs/features/shared-volumes-with-NFS.md

```diff
@@ -0,0 +1,207 @@
+# Supporting shared volumes with vSphere Docker Volumes
+
+
+## Requirements
+* Developer should able to use externally managed NFS share as “shared docker volume” using “vsphere” volume driver.
+* This allows them to manage NFS directories as docker volumes using Docker APIs, Compose etc 
+* Extend the vSphere Docker Volume plugin to support shared volumes
+* Volumes are accessible on multiple docker hosts running on  multiple ESX hosts in a Docker swarm
+* Allow user to specify what remote dirs are managed by the plugin
+* Support all current docker volume plugin API for shared volumes
+
+##  Assumptions
+* Administrator sets up shares on NFS server(s) and ensures Docker Hosts has permissions to mount and modify the remote dirs.
+* Developer responsible for (one-time) plugin configuration.
+* 
+
+
+## vSphere Docker Volume Plugin Drivers – User experience
+
+  1. Setup
+      a. Set up Plugin Config
+      b. Restart plugin
+  2. Using shared volumes
+      a. docker volume create …
+      b. docker volume list …
+      c. docker volume inspect …
+      d. docker run …
+      e. docker volume rm …
+
+
+# vSphere Docker Volume Plugin Drivers - Background
+
+* Docker Volume Plugin supports a framework to implement storage specific drivers
+* Drivers implement a common interface
+   - Supports the volume plugin API of Docker volume plugins
+   - Mount, Unmount, Get, Capabilities, Create, Remove
+* Selection of driver via plugin configuration
+* Easily extendable to any storage backing for container volumes
+* A new driver can be written to support a specific storage type
+* Presently, two drivers are supported
+    -  Provide VMDK backed Docker volumes on platform specific storage
+    - vSphere – provide Docker volumes vSAN, VVOL, VMFS, NFS
+    - Photon – vSAN
+* Drivers work with endpoints (currently local-ESX and Photon) to manage volumes
+    - vSphere – works with the local ESX host to provision VMDKs
+    - Photon – works with a Photon Controller endpoint to provision VMDKs
+
+
+# vSphere-cloud driver
+
+* Add a new “vsphere-cloud” (or better name) driver
```

Comment:
From review of this looks like a new driver may imply less change to existing code but more drivers may be unnecessary. "vsphere" driver is ok to use for all native vsphere storage and other storage backends.